tests/mock_network/mod.rs

```diff
@@ -65,6 +66,55 @@ fn nodes_with_prefix_mut<'a>(
         .filter(move |node| prefix.matches(&node.name()))
 }
 
+pub fn count_sections_members_if_split(nodes: &[TestNode]) -> BTreeMap<Prefix<XorName>, usize> {
+    let mut counts = BTreeMap::new();
+    for pfx in nodes.iter().map(node_prefix_if_split) {
+        *counts.entry(pfx).or_default() += 1;
+        let _ = counts.entry(pfx.sibling()).or_default();
```

Comment:
why need to create an entry for sibling here?
we already used `entry().or_default` above

Reply:
Because it failed the test :-)
The problem is if the first 3 nodes happen to be in the same sub prefix the map only contain one prefix. with this it contain one sub prefix with 3 and the other sub-prefix with 0.

Reply:
> Because it failed the test ...  the other sub-prefix with 0

hmm...  this means the map has to be `prefix symmetric`, right?
I think the test failure is due to not catering the `non-symmetric` case. 
i.e. not generating new nodes evently.

I won't insisit to cartering for it, but better leave a comment here to remind us why this is required (and maybe need to cover it later on) ?

Reply:
This is because of how we use the map, if the map does not cover the full address space we have a problem. The reason we have a problem is when we create a new node, we look for which sub-prefix it is in and if there is space. If there is no sub-prefix for it, then it cannot be added. (In the degenerate case I mentioned, 3 node in same sub-prefix 0, you can never add any node to sub-prefix 1, and it never split).